### ISSUE TITLE
Playing multiple mp3s

### ISSUE THREAD

**Issue by zsarosi** [nextcloud/ios#312]
Hi
I am trying to play a list of mp3 from a folder in iOS Nextcloud client. Currently, I can only play them one-by-one individually, by clicking on every single one. Is there a possibility to play all mp3s from a folder continuously? When playing a single mp3 I can see the options "repeat" and "shuffle" at top right corner, which would make no sense, if only a single file can be played, so i guess i am missing something.
thanks

**Comment by wizdude** [comment]
+1

i would love the music playing capability of the ios client to be enhanced. i use other apps such as "cloudbeats" or "cloud music" to sync and play my music. there is no reason why this couldn't be done in this ios client.

my main issue with playing music is i lose control. i click on a music file to play, it starts playing and if i go to another page in the ios client to do other work there is no way to return back to the audio controls. i have to flick up and use the native ios controls to pause or stop the music playing.

i would *love* to be able to use this as my primary music player.

cheers, wizdude.

**Comment by veddox** [comment]
This would be fantastic to have! I have all my music on my Nextcloud server, and don't really want to go through the hassle of keeping it in sync with the standard Apple music app... The Nextcloud web music app recently added this capability (https://github.com/owncloud/music/issues/641) and it would be great to have it in the mobile app too :-)

**Comment by localguru** [comment]
+1

**Comment by rehste** [comment]
Do we still have a chance of implementing this topic in the coming months? There are apps like "CloudBeats" but I think it's better and safer to have everything integrated in one app. Many users would certainly be very happy....
+1

**Comment by sleif** [comment]
+1

**Comment by tobzsc** [comment]
I have exactly the same problem. :( Would be nice to see this!

+1

**Comment by veddox** [comment]
I've had a look at the source code, but I've never done iOS development and don't have a Mac to compile on... It shouldn't be a big change IMO (just jump to the next mp3 in a folder after playback finishes), but I'm afraid I'm the wrong person to implement it :slightly_frowning_face:

**Comment by marinofaggiana** [comment]
I put it on the list ToDo ;)

**Comment by AInteriorB** [comment]
Great, @marinofaggiana, I think an easy music player that is capable of playing a whole directory using the native iOS player (thereby playing in background/looked iPhones is possible (#534)) would be one of the biggest enhancements of the app! Thanks for your great work!

**Comment by KarimJedda** [comment]
Any update on this?

**Comment by KarimJedda** [comment]
@veddox Well I found a solution but it's absolutely bonkers and hacky. Instead of AVPlayer I used AVQueuePlayer and populated the queue with the downloaded songs. For the life of me I could not add a "Next" or "Previous" song to skip songs, I scoured the whole interwebs while my cat was watching and adding a next/previous functionality seems to be some kind of dark art. 
I managed to autoplay the next song using the NotificationCenter "Avqueplayeritemdidfinish.." trigger (or whatever that is, I'm not a Swift dev) and adding appDelegate.player.advanceToNextItem() instead of player.seek(CMTime 0). 
@marinofaggiana can I support you in any way pushing this forward?

**Comment by veddox** [comment]
@KarimJedda Yes, I tried hacking something up myself, but got stuck just like you did. (Although it didn't help that I've never done iOS development before ;-) )

I would be very happy if somebody could take a crack at this, I still sorely miss the feature :-( Perhaps we could set a bounty on it somewhere?

**Comment by tarator** [comment]
+1

**Comment by kbaegis** [comment]
Still open, 4 years later :D

Do we even know if this is feasible?
